### PR TITLE
CANS-291: Users Can Only Edit Assessments in their County

### PR DESCRIPTION
**Pull request by DmitriyRudenko**
## Tests

- [x] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by LeonidMarushevskyi** [comment]
Jenkins retest this please

**Comment by LeonidMarushevskyi** [comment]
Jenkins retest this please

**Comment by LeonidMarushevskyi** [comment]
No related changes are removed from PR

File: app/javascript/Application/components/Assessment/AssessmentContainer.test.js [hunk 1]
```diff
@@ -25,13 +25,16 @@ const mountWithRouter = async component => mount(<MemoryRouter initialEntries={[
 
 describe('<AssessmentContainer />', () => {
   describe('init AssessmentContainer', () => {
-    describe('page layout', () => {
+    describe('page layout', async () => {
```
Comment (edgibbs) [inline app/javascript/Application/components/Assessment/AssessmentContainer.test.js]: So this took quite a bit of digging, but the specs here are not being run based on async being setup in a describe block which apparently is syntactically allowed, but not supported by jest.  So the 4 specs associated with this block were not running.  After some debugging I was able to get them to run again with the following code:

```javascript
    describe('page layout', () => {
      const props = { 
        location: { childId: 10 },
        match: { params: { id: 1 } },
      };  

      beforeEach(() => {
        jest.spyOn(ClientService, 'fetch').mockReturnValue(Promise.resolve(childInfoJson));
        jest.spyOn(SecurityService, 'checkPermission').mockReturnValue(Promise.resolve(true));
        jest.spyOn(AssessmentService, 'fetch').mockReturnValue(Promise.resolve(assessment));
      }); 

      it('renders with 1 <PageInfo /> component', () => {
        const wrapper = shallow(<AssessmentContainer {...props} />);
        expect(wrapper.find(PageInfo).length).toBe(1);
      });

      it('renders with 1 <AssessmentFormHeader /> component', () => {
        const wrapper = shallow(<AssessmentContainer {...props} />);
        expect(wrapper.find(AssessmentFormHeader).length).toBe(1);
      });

      it('stuff renders with 1 <Typography /> component with submit instructions', async () => {
        const wrapper = shallow(<AssessmentContainer {...props} />);
        await wrapper.instance().componentDidMount();
        expect(wrapper.find(Typography).prop('className')).toBe('submit-validation-message');
      });

      it('renders with 1 <AssessmentFormFooter /> component', () => {
        const wrapper = shallow(<AssessmentContainer {...props} />);                                                                                                                           
        expect(wrapper.find(AssessmentFormFooter).length).toBe(1);
      }); 
    }); 
```

Note the third spec had to do an await on the componentDidMount() due to the fact that the logic is set there and it does not happen in time to set the render body to the right state.

**Pull request merged**